### PR TITLE
Handle metav1duration upstream issue

### PR DESCRIPTION
**Pull request by rohitsakala**
Please look into the issue for more information. 

- Replaceing metav1.Duration with time.Duration due to an upstream issue and It is advised to not use [metav1.Duration](https://github.com/rancher/rancher/pull/45062/files#diff-b335c5d6572cc7826cac4d401e0534b1830b0744fdaaa5e22cee2dd03fd13bd7R28) due to an [upstream issue](https://github.com/kubernetes/apimachinery/issues/131).

### Next Steps 

- [x] Update UI issue with string instead of metav1.Duration.

### CODE REVIEW

File: pkg/catalogv2/oci/client.go [hunk 1]
```diff
@@ -228,13 +228,31 @@ func (o *Client) getAuthClient() (*http.Client, error) {
 		if o.exponentialBackOffValues.MaxRetries > 0 {
 			retryPolicy.MaxRetry = o.exponentialBackOffValues.MaxRetries
 		}
-		if o.exponentialBackOffValues.MaxWait != nil {
-			retryPolicy.MaxWait = o.exponentialBackOffValues.MaxWait.Duration
+		if o.exponentialBackOffValues.MaxWait != "" {
+			maxWait, err := time.ParseDuration(o.exponentialBackOffValues.MaxWait)
+			if err != nil {
+				return nil, err
```
Comment (pjbgf) [inline pkg/catalogv2/oci/client.go:234]: For improved troubleshooting:
```suggestion
				return nil, fmt.Errorf("failed to parse exponentialBackOffValues.MaxWait: %w", err)
```

File: pkg/catalogv2/oci/client.go [hunk 2]
```diff
@@ -228,13 +228,31 @@ func (o *Client) getAuthClient() (*http.Client, error) {
...
+			}
+			retryPolicy.MaxWait = maxWait
 		}
-		if o.exponentialBackOffValues.MinWait != nil {
-			retryPolicy.MinWait = o.exponentialBackOffValues.MinWait.Duration
+		if o.exponentialBackOffValues.MinWait != "" {
+			minWait, err := time.ParseDuration(o.exponentialBackOffValues.MinWait)
+			if err != nil {
+				return nil, err
```
Comment (pjbgf) [inline pkg/catalogv2/oci/client.go:241]: For improved troubleshooting:
```suggestion
				return nil, fmt.Errorf("failed to parse exponentialBackOffValues.MinWait: %w", err)
```

File: pkg/catalogv2/oci/client.go [hunk 3]
```diff
@@ -228,13 +228,31 @@ func (o *Client) getAuthClient() (*http.Client, error) {
...
+			}
+			retryPolicy.MinWait = minWait
 		}
 	}
+	// The minimum duration should be atleast 1 second
+	if retryPolicy.MinWait < 1*time.Second {
+		return nil, errors.New("minWait should be atleast 1 second")
```
Comment (pjbgf) [inline pkg/catalogv2/oci/client.go:248]:
```suggestion
		return nil, errors.New("minWait should be at least 1 second")
```

File: pkg/catalogv2/oci/client.go (hunk 1 above)
Comment (rohitsakala) [reply pkg/catalogv2/oci/client.go:234]: Done.

File: pkg/catalogv2/oci/client.go (hunk 3 above)
Comment (rohitsakala) [reply pkg/catalogv2/oci/client.go:248]: Done.

**Review by pjbgf — APPROVED** [review]

**Comment by pjbgf** [comment]
@rohitsakala this needs a rebase to fix the CI issue.

**Pull request merged**